nucypher/blockchain/eth/networks.py

```diff
@@ -1,57 +1,75 @@
-class NetworksInventory:  # TODO: See #1564
-
-    MAINNET = "mainnet"
-    LYNX = "lynx"
-    IBEX = "ibex"  # this is required for configuration file migrations (backwards compatibility)
-    ETH = "ethereum"
-    TAPIR = "tapir"
-    ORYX = "oryx"
-
-    # TODO: Use naming scheme to preserve multiple compatibility with multiple deployments to a single network?
-    POLYGON = 'polygon'
-    MUMBAI = 'mumbai'
-
-    UNKNOWN = 'unknown'  # TODO: Is there a better way to signal an unknown network?
-    DEFAULT = MAINNET
-
-    __to_chain_id_eth = {
-        MAINNET: 1,  # Ethereum Mainnet
-        IBEX: 5,  # this is required for configuration file migrations (backwards compatibility)
-        LYNX: 5,  # Goerli
-        TAPIR: 5,
-        ORYX: 5
-    }
-    __to_chain_id_polygon = {
-        # TODO: Use naming scheme?
-        POLYGON: 137,    # Polygon Mainnet
-        MUMBAI: 80001,   # Polygon Testnet (Mumbai)
-    }
-
-    ETH_NETWORKS = tuple(__to_chain_id_eth.keys())
-    POLY_NETWORKS = tuple(__to_chain_id_polygon.keys())
-
-    NETWORKS = ETH_NETWORKS + POLY_NETWORKS
-
-    class UnrecognizedNetwork(RuntimeError):
-        pass
+from enum import Enum
+from typing import List, NamedTuple
 
-    @classmethod
-    def get_ethereum_chain_id(cls, network):
-        try:
-            return cls.__to_chain_id_eth[network]
-        except KeyError:
-            raise cls.UnrecognizedNetwork(network)
+from nucypher.config.constants import TEMPORARY_DOMAIN
+
+
+class ChainInfo(NamedTuple):
+    id: int
+    name: str
+
+
+class EthChain(ChainInfo, Enum):
+    MAINNET = ChainInfo(1, "mainnet")
+    GOERLI = ChainInfo(5, "goerli")
+    SEPOLIA = ChainInfo(11155111, "sepolia")
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class PolygonChain(ChainInfo, Enum):
+    POLYGON = ChainInfo(137, "polygon")
+    MUMBAI = ChainInfo(80001, "mumbai")
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class TACoNetwork(NamedTuple):
+    name: str
+    eth_chain: EthChain
+    polygon_chain: PolygonChain
+
+    def is_testnet(self) -> bool:
+        return self.eth_chain != EthChain.MAINNET
+
+
+class UnrecognizedNetwork(RuntimeError):
+    """Raised when a provided network name is not recognized."""
+
+
+class NetworksInventory:
+    MAINNET = TACoNetwork("mainnet", EthChain.MAINNET, PolygonChain.POLYGON)
+    # Testnets
+    ORYX = TACoNetwork("oryx", EthChain.GOERLI, PolygonChain.POLYGON)
+    LYNX = TACoNetwork("lynx", EthChain.GOERLI, PolygonChain.MUMBAI)
+    TAPIR = TACoNetwork("tapir", EthChain.GOERLI, PolygonChain.MUMBAI)
+    # TODO did Ibex even use a PolyNetwork?
+    IBEX = TACoNetwork(
+        "ibex", EthChain.GOERLI, PolygonChain.MUMBAI
+    )  # this is required for configuration file migrations (backwards compatibility)
```

Comment:
No, it did not.  Perhaps use a sentinel value instead.  This could be as simple as `None` but you could use some kind of constant.

Reply:
✔️  - used `None`. It's unclear to me that we need an ibex entry at all. I'm just adhering to the comment which was there from before.